src/covid_model_seiir_pipeline/forecasting/specification.py

```diff
@@ -1,17 +1,67 @@
 from dataclasses import dataclass, field
 import itertools
-from typing import Dict, Tuple, Union
+from typing import Dict, NamedTuple, Tuple, Union
 
 from covid_model_seiir_pipeline.utilities import Specification, asdict
+from covid_model_seiir_pipeline.workflow_tools.specification import TaskSpecification, WorkflowSpecification
 
-# TODO: Extract these into specification, maybe.  At least allow overrides
-#    for the queue from the command line.
-FORECAST_RUNTIME = 15000
-FORECAST_MEMORY = '5G'
-POSTPROCESS_MEMORY = '150G'
-FORECAST_CORES = 1
-FORECAST_SCALING_CORES = 26
-FORECAST_QUEUE = 'd.q'
+
+class __ForecastJobs(NamedTuple):
+    scaling: str = 'scaling'
+    forecast: str = 'forecast'
+    sentinel: str = 'sentinel'
+    resample: str = 'resample'
+    postprocess: str = 'postprocess'
+
+
+FORECAST_JOBS = __ForecastJobs()
+
+
+class ScalingTaskSpecification(TaskSpecification):
+    """Specification of execution parameters for beta scaling tasks."""
+    default_max_runtime_seconds = 5000
+    default_m_mem_free = '5G'
+    default_num_cores = 26
+
+
+class ForecastTaskSpecification(TaskSpecification):
+    """Specification of execution parameters for beta forecasting tasks."""
+    default_max_runtime_seconds = 15000
+    default_m_mem_free = '5G'
+    default_num_cores = 1
+
+
+class SentinelTaskSpecification(TaskSpecification):
+    """Specification of execution parameters for workflow join tasks."""
+    default_max_runtime_seconds = 1000
+    default_m_mem_free = '1G'
+    default_num_cores = 1
+
+
+class ResampleTaskSpecification(TaskSpecification):
+    """Specification of execution parameters for draw resample mapping tasks."""
+    default_max_runtime_seconds = 5000
+    default_m_mem_free = '50G'
+    default_num_cores = 26
+
+
+class PostprocessingTaskSpecification(TaskSpecification):
+    """Specification of execution parameters for postprocessing tasks."""
+    default_max_runtime_seconds = 15000
+    default_m_mem_free = '150G'
+    default_num_cores = 26
+
+
+class ForecastWorkflowSpecification(WorkflowSpecification):
+    """Specification of execution parameters for forecasting workflows."""
+
+    tasks = {
+        FORECAST_JOBS.scaling: ScalingTaskSpecification,
+        FORECAST_JOBS.forecast: ForecastTaskSpecification,
+        FORECAST_JOBS.sentinel: SentinelTaskSpecification,
+        FORECAST_JOBS.resample: ResampleTaskSpecification,
+        FORECAST_JOBS.postprocess: PostprocessingTaskSpecification,
+    }
```

Comment:
what is this mapping for? It looks like expected types but it's a class constant

Reply:
It get's used here: https://github.com/ihmeuw/covid-model-seiir-pipeline/pull/123/files#diff-b3cc748ab1833dfcf96f29d680ca4968ab2352f8ab0fac8986eb649d074ad191R123-R139
to turn the TaskSpecification subclasses (which are essentially bundles of sensible default values) into TaskSpecification objects which resolve inputs from the specification file with the defaults and do a little validation.

Reply:
I'm doing this all with class variables to mirror the way other specifications are constructed out of data classes with a very similar interface.